tests/client.py

```diff
@@ -778,6 +780,103 @@ def create(self, stream, company_ids=[], subscriptions=[], times=1):
 
+    def create_custom_contact_properties(self):
+        """Create custom contact properties of all the types"""
+
+        url = f"{BASE_URL}/properties/v1/contacts/properties"
+        data = []
+        property = {
+            "name": "custom_string",
+            "label": "A New String Custom Property",
+            "description": "A new string property for you",
+            "groupName": "contactinformation",
+            "type": "string",
+            "fieldType": "text",
+            "formField": True,
+            "displayOrder": 6,
+            "options": [
+            ]
+        }
+        data.append(deepcopy(property))
+
+        property = {
+            "name": "custom_number",
+            "label": "A New Number Custom Property",
+            "description": "A new number property for you",
+            "groupName": "contactinformation",
+            "type": "number",
+            "fieldType": "text",
+            "formField": True,
+            "displayOrder": 7,
+            "options": [
+            ]
+        }
+        data.append(deepcopy(property))
+
+        property = {
+            "name": "custom_date",
+            "label": "A New Date Custom Property",
+            "description": "A new date property for you",
+            "groupName": "contactinformation",
+            "type": "date",
+            "fieldType": "text",
+            "formField": True,
+            "displayOrder": 9,
+            "options": [
+            ]
+        }
+        data.append(deepcopy(property))
+
+        property = {
+            "name": "custom_datetime",
+            "label": "A New Datetime Custom Property",
+            "description": "A new datetime property for you",
+            "groupName": "contactinformation",
+            "type": "datetime",
+            "fieldType": "text",
+            "formField": True,
+            "displayOrder": 10,
+            "options": [
+            ]
+        }
+        data.append(deepcopy(property))
+
+        property = {
+            "name": "multi_pick",
+            "label": "multi pick",
+            "description": "multi select picklist test",
+            "groupName": "contactinformation",
+            "type": "enumeration",
+            "fieldType": "checkbox",
+            "hidden": False,
+            "options": [
+              {
+                "label": "Option A",
+                "value": "option_a"
+              },
+              {
+                "label": "Option B",
+                "value": "option_b"
+              },
+              {
+                "label": "Option C",
+                "value": "option_c"
+              }
+            ],
+            "formField": True
+        }
+        data.append(deepcopy(property))
+        # generate a contacts record
+
+        for current_data in data:
+            try:
+                response = self.post(url, current_data)
+                LOGGER.info("response is %s", response)
+            except Exception as e:
```

Comment:
This Exception is too broad.  We should be a specific with exceptions as possible as this will swallow any exception, some not intended here.

Also, I'm not sure why you would get an exception if data already exists instead of getting a 4xx response code?

Reply:
yeah, it is a 409, which was causing the test to fail. I'll handle it specifically instead of a broader one.

Reply:
I think we miscommunicated.  I meant in the `except Exception as e` we should actually change the word `Exception` to a more specific exception.  I don't know what kind it is but based on the 409 maybe it is an RequestException or HTTPError?  However, I didn't think a 409 would actually raise an exception.  If you are trying to do this until you get a successful status code maybe another way would be better. Let's talk about what you are trying to achieve to determine the best way to go about it.

Reply:
the post call returns a response code of 409 and in post method the exception is raised. That's why I am handling it here. I can change the exception to a specific one to make it more readable.